### DISCUSSION TITLE
Can I add a cmake option when installing the port?

### DISCUSSION THREAD

**Discussion by hhxdestiny** [microsoft/vcpkg#35347] [Q&A] [answered]
When I install ncnn using vcpkg, I expect to open NCNN_ VULKAN in CMakeLists.txt option.

![image](https://github.com/microsoft/vcpkg/assets/44420757/73b98871-330c-4df4-9e41-e576c7d7f5d7)

So I re edited vcpkg/ports/ncnn/portfile.cmake , , added -DNCNN_VULKAN=ON under the Options section of vcpkg_cmake_configure. Then I uninstalled ncnn and reinstalled it, resulting in the following error

```bash
CMake Error at CMakeLists.txt:587 (message):
  The submodules were not downloaded! Please update submodules with "git
  submodule update --init" and try again.


-- Configuring incomplete, errors occurred!
```

![image](https://github.com/microsoft/vcpkg/assets/44420757/9d4d3878-b062-472b-b238-7e5f2466a01a)

What is the correct approach?

**Comment by dg0yt** [comment, marked as answer]
It looks like the desired feature requires code from another repository. vcpkg doesn't directly support git submodules. (Direct git repo access doesn't allow asset caching.) So you will have to extend the portfile to explicitly fetch the other repo and move it to the submodule location.
